AsyncDisplayKit/Layout/ASDimension.h

```diff
@@ -12,56 +12,183 @@
 #import <UIKit/UIKit.h>
 #import <AsyncDisplayKit/ASBaseDefines.h>
 
-/** A dimension relative to constraints to be provided in the future. */
-typedef NS_ENUM(NSInteger, ASRelativeDimensionType) {
+#define ASPointsAreValidForLayout(x) ((isnormal(x) || x == 0.0) && x >= 0.0 && x < (CGFLOAT_MAX / 2.0))
+#define ASCGSizeIsValidForLayout(x) (ASPointsAreValidForLayout(x.width) && ASPointsAreValidForLayout(x.height))
```

Comment:
Can we define these as static inline functions and gain the advantage of specifying the parameter types?